### ISSUE TITLE
balance: cap remorts, HP & Stamina

### ISSUE THREAD

**Issue by Tocharaeh** [4dmud/mud#22]
- We need something to be able to sweep pfiles and locate those with HP great than 7k hp, and then reset their HP to the max HP appropriate for their class.

- Should setup HP caps to be class specific:
  >Casters -  4500hp
  >Rogues -  5500hp
  >Fighters - 6500hp
Give GMs a +500hp bonus to that max.

**Comment by lfaoro** [comment]
I can make a script that parses all pfiles and output any pfile over the provided integer, could be used like this:
```bash
./psearch Hit: 5000
Name: Imm has Hit: 5700
Name: Toch has Hit: 8000
...etc
```

**Comment by Rynald4d** [comment]
If we code it in so the cap is forced at login we don't have to worry about it anymore. Should we want to change the cap then that would be an easy change as well.

**Comment by lfaoro** [comment]
agreed, let's code it in then

**Comment by Tocharaeh** [comment]
Great idea(s) :)

**Comment by J00bei** [comment]
I'd would recommend just limiting the Bonus HP from remorts to a max of 1k remorts putting a cap on this is really only targeting 3 players, myself, Vryce and Zi I feel like I've repeated myself many times on this.

I disagree on bonus of 500 for gm also. The current imbalance of HP is only us 3 players and it is due to this remort bonus and setting it to a 1k remort max bonus will fix this issue and give other players an incentive to actually level.

**Comment by J00bei** [comment]
Just to add I am quite sure, me, Vryce and Zi would have no issue remorting again to cause our hp to reset values that should be balanced to these new changes.

**Comment by Vryyce** [comment]
What are you guys trying to do here? **CAP THE REMORTS**. Dont go messing around in the Pfiles manually or via whateverthefuck. If youve got questions about how this can horribly go wrong ask Kitolani/Eden. 
 Ive asked multiple times to have my file reset one year to solve this. 
Rynald fixed the infinite money glitch. Now reset myself, Jubei and Zi pfiles FROM THE BACKUP! one year, CAP THE REMORTS and the problem is solved

**Comment by lfaoro** [comment]
what are you suggesting @Vryyce to cap the remorts?
we're talking to cap the max HP not the remorts

**Comment by J00bei** [comment]
> 
> 
> what are you suggesting @Vryyce to cap the remorts? we're talking to cap the max HP not the remorts, you should still be able to remort as much as you want and change classes.

Did you look at the code regarding why there is such a bonus to hp?

Specifically the one that allows to get such high levels,  look at the remort.cpp

Namely this section...

REMORTS ( ch ) ++;
    GET_MAX_HIT ( ch ) = 30 + remorts;

Putting a maximum cap on this value will remort solve the hp issues for the high remort characters whenever we were to remort next, if i am correct.

We are saying there is no need to cap the max HP, just cap this to have a max bonus of 1000 remorts instead of unlimited.

**Comment by lfaoro** [comment]
I understand now, I'd cap it at GM level, which is about 50 remorts.

**Comment by Vryyce** [comment]
@lfaoro There should be some benefit to additional remorts. 500 or 1000 remorts is still very minimal gains but makes it worthwhile to achieve. HP, mana, moves and stamina are all tied to the remort bonus so by capping the max remorts you can cap all of those stats at the same time

**Comment by J00bei** [comment]
Well I would cap the bonus  at 1000 remorts, which would give a swing of 1k extra HP for those that wanted to progress their characters more via remorting, gaining a slight edge over other players that would rather sit in recall. 

From my experience a ceiling of 1000 remorts is reachable by any player that would want to dedicate their time to get a edge like this. 

It also gives an incentive to players that would promote getting out there and killing stuff, if they are wanting to be competitive.

**Comment by lfaoro** [comment]
I don't think we solve anything with 1000+30 remort bonus, as player will still get over the limits suggested by @Tocharaeh 

1,000 remorts means = 1000+30 = 1030HP bonus, which still gets users a 1k HP bonus over anyone else.

if it wasn't for remorting using gold who in his sane mind would remort 1,000 times by killing mobs?

**Comment by Vryyce** [comment]
I disagree that 1000 remorts wouldn't solve anything.
 I think it would completely solve the issue while still giving players an achievable goal if they chose to grind that much because I personally had 450 remorts on mobs only. 
These issues only showed themselves when we gained 20,000, 40,000 or Zi's 100,000 remort levels.

**Comment by lfaoro** [comment]
450 remorts on mob only still gives you a 530HP bonus more than what you've grinded for, I believe it's fair.

**Comment by lfaoro** [comment]
As reported by @Vryyce high remort players can walk through SPACE without any challenge, thanks to their infinite Stamina, simplifying space exploration against other players that are required to use ships to move. Adding the same cap for stamina as HP.

**Comment by J00bei** [comment]
What exactly is wrong with having another goal to work towards. 

Who are you to say what new players can and cant do, I suspect this pushback from you is purely coming from a pvp perspective. 

Which pvp is currently dead and even if it wasn't if you think 1k hp will make a difference I will let you know right now it won't. Vryce and I know PVP and PVE off the back of our hands no matter how much you look at the code there are things you won't be able to account for. We know because we have actively played here for the last 8 years at the highest level.

As someone who also remorted 430 times on mobs. 

'450 remorts on mob only still gives you a 530HP bonus more than what you've grinded for, I believe it's fair.' 

It's not fair because not everyone did remort 400 times plus on mobs so why would a global 500 remort bonus for every gm be fair when they didnt earn it.

If you want to add a gm bonus of 500 then don't incorporate it into this remort bonus because it isn't the same thing.

While I wouldn't totally be opposed with a 500 hp bonus on gm. I would be opposed to scrapping the idea of providing a remort bonus capped at about 1k.

**Comment by lfaoro** [comment]
Guys, I'm not here to argue about this change, I thought we've already discussed on Discord and we all agreed that we have to start doing something to bring back the balance to 4D.

The changes I've submitted are not merged in the code, they're just pending changes, we can close them and leave things as they are. I'm not here to fight with you over these issues. 

You want them they go in, you don't want them we throw them away. No worries. Discuss with the Imms, not with me. I believe they have the final say.

Cheers
